services/checks/checks.go

```diff
@@ -419,6 +433,98 @@ func (s *Service) EnableChecks(checkNames []string) error {
 	return nil
 }
 
+// ChangeInterval changes a check's interval to the value received from the UI.
+func (s *Service) ChangeInterval(params map[string]check.Interval) error {
+	checkMap := s.GetAllChecks()
+	if len(checkMap) == 0 {
+		return errors.New("no checks loaded")
+	}
+
+	for name, interval := range params {
+		c, ok := checkMap[name]
+		if !ok {
+			return errors.Errorf("check: %s not found", name)
+		}
+		c.Interval = interval
+
+		// since we re-run checks at regular intervals using a call
+		// to s.StartChecks which in turn calls s.collectChecks
+		// to load/download checks, we must persist any changes
+		// to check intervals in the DB so that they can be re-applied
+		// once the checks have been re-loaded on restarts.
+		e := s.db.InTransaction(func(tx *reform.TX) error {
+			cs, err := models.FindCheckStateByName(s.db.Querier, c.Name)
+			// record interval change for the first time.
+			if err == reform.ErrNoRows {
+				cs, err := models.CreateCheckState(s.db.Querier, c.Name, models.Interval(c.Interval))
+				if err != nil {
+					return err
+				}
+				s.l.Debugf("Saved interval change for check: %s in DB", cs.Name)
+				return nil
+			}
+
+			// update existing interval change.
+			if cs != nil {
+				cs, err := models.ChangeCheckState(s.db.Querier, c.Name, models.Interval(c.Interval))
+				if err != nil {
+					return err
+				}
+				s.l.Debugf("Updated interval change for check: %s in DB", cs.Name)
+				return nil
+			}
+			return err
+		})
+		if e != nil {
+			return e
+		}
+
+		s.updateCheck(c)
```

Comment:
I suggest to include it into transaction

Reply:
done